Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.cpp

```diff
@@ -473,6 +474,26 @@ void WebResourceLoadStatisticsStore::requestStorageAccess(RegistrableDomain&& su
     });
 }
 
+void WebResourceLoadStatisticsStore::setLoginStatus(RegistrableDomain&& domain, IsLoggedIn loggedInStatus, CompletionHandler<void()>&& completionHandler)
+{
+    ASSERT(RunLoop::isMain());
+
+    if (loggedInStatus == IsLoggedIn::Yes)
+        m_loginStatus.set(domain, loggedInStatus);
+    else
+        m_loginStatus.remove(domain);
```

Comment:
Maybe we should replace if/else with just `m_loginStatus.set(domain, loggedInStatus);`.

Reply:
But we don't want to assign logged-out to a domain, right? We just want to remove them from the hash map bc every user is logged-out by default.

Reply:
It would be the same behavior, it would just save us a few lines of code. Though this hash map isn't a permanent solution, so it doesn't matter much. Up to you.

Reply:
I think removing entries from the map when possible is cleaner. We wouldn't want a forever-growing hash map.